homeassistant/components/smarttub/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Config flow to configure the SmartTub integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, CONF_SCAN_INTERVAL
+from homeassistant.core import callback
+
+from .const import DEFAULT_SCAN_INTERVAL, DOMAIN  # pylint: disable=unused-import
+from .controller import SmartTubController
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_EMAIL): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SmartTubConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """SmartTub configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is not None:
+            controller = SmartTubController(self.hass)
+            account_id = await controller.get_account_id(
+                user_input[CONF_EMAIL], user_input[CONF_PASSWORD]
+            )
+            if account_id is not None:
+                await self.async_set_unique_id(account_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_EMAIL], data=user_input
+                )
+
+            errors["base"] = "invalid_auth"
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_reauth(self, user_input=None):
+        """Handle a flow initiated by failed auth."""
+        return await self.async_step_user(user_input)
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
+                        default=self.config_entry.options.get(
+                            CONF_SCAN_INTERVAL,
+                            DEFAULT_SCAN_INTERVAL,
+                        ),
+                    ): int,
```

Comment:
Is there a reason that this needs to be adjustable?

Reply:
Not especially. I wasn't sure what interval would be most appropriate, and a lot of integrations seem to make it configurable, so I followed suit. Would you prefer it be hardcoded?

Reply:
As developers we should choose a reasonable default and not expose to the control unless its strictly needed as users tend to set it too low and get integrations banned from cloud services.

Reply:
I removed this, and since it was the only option, I removed the options flow and test